csharp/src/Ice/Properties.cs

```diff
@@ -3,75 +3,82 @@
 #nullable enable
 
 using Ice.Internal;
+using System.Collections.Immutable;
+using System.ComponentModel;
 using System.Globalization;
 using System.Text;
 using System.Text.RegularExpressions;
 
 namespace Ice;
 
 /// <summary>
-/// A property set used to configure Ice and Ice applications. Properties are key/value pairs, with both keys and
-/// values being strings.
+/// Represents a set of properties used to configure Ice and Ice-based applications. A property is a key/value pair,
+/// where both the key and the value are strings. By convention, property keys should have the form
+/// <c>application-name>[.category[.sub-category]].name</c>.
 /// </summary>
+/// <remarks>This class is thread-safe: multiple threads can safely read and write the properties without their own
+/// synchronization.</remarks>
```

Comment:
I will remove the without synchronization, that is imply when you say is thread-safe